app/models/admin/reports/duplicate_approved_claims.rb

```diff
@@ -0,0 +1,222 @@
+      # `teacher_reference_number`.
+      def claimable_policy_mapping(policy)
+        policy.policies_claimable.map do |other_policy|
+          shared_matching_attributes = policy.eligibility_matching_attributes.select do |attribute_group|
+            attribute_group.all? do |attr|
+              other_policy::Eligibility.column_names.include?(attr)
+            end
+          end
+
+          [other_policy, shared_matching_attributes]
+        end.to_h.reject { |_, matching_attrs| matching_attrs.empty? }
+      end
+
+      # building_society_roll_number is no longer used, so is always null
+      # we only check the number and sort code when determining duplicates.
+      def claim_matching_attributes
+        Claim::MatchingAttributeFinder::CLAIM_ATTRIBUTE_GROUPS_TO_MATCH.map do |attr_group|
+          attr_group.without("building_society_roll_number")
+        end
+      end
+
+      def duplicates_by_attributes
+        # Limit the claims we're looking at
+        current_claims = <<~SQL
+          WITH current_claims AS (
+            SELECT claims.id, #{claim_matching_attributes.flatten.join(", ")}
+            FROM claims
+            JOIN decisions ON claims.id = decisions.claim_id
+            WHERE claims.academic_year = '#{academic_year}'
+            AND decisions.undone = false
+            AND decisions.result = 0
+          )
+        SQL
+
+        # Make sure to have indexes for the columns we're querying!
+        filter = claim_matching_attributes.flat_map do |attribute_group|
+          join_condition = attribute_group.map do |attr|
+            if Claim.column_for_attribute(attr).type == :string
+              "LOWER(current_claims.#{attr}) = LOWER(other_claims.#{attr})"
+            else
+              "current_claims.#{attr} = other_claims.#{attr}"
+            end
+          end.join(" AND ")
+
+          <<~SQL
+            SELECT current_claims.id, other_claims.id AS other_claim_id
+            FROM current_claims
+            JOIN current_claims other_claims
+            ON #{join_condition}
+            WHERE current_claims.id != other_claims.id
+          SQL
+        end.join("\nUNION\n")
+
+        query = current_claims + "\n" + filter
+
+        ActiveRecord::Base.connection.execute(query).map(&DuplicateClaimRow.method(:new))
+      end
+
+      class ClaimPresenter
+        include Admin::ClaimsHelper
+
+        def initialize(claim)
+          @claim = claim
+        end
+
+        def to_a
+          [
+            claim.reference,
+            claim.full_name,
+            claim.eligibility.try(:teacher_reference_number),
+            I18n.t("#{claim.policy.locale_key}.policy_acronym"),
+            claim.award_amount,
+            status(claim),
+            claim.decisions.last.created_at.to_date,
+            claim.decisions.last.created_by&.full_name
+          ]
+        end
+
+        private
+
+        attr_reader :claim
+      end
+    end
+  end
+end
```

Comment:
@kenfodder @slorek instead of doing this gnarly query we could:
* add a new claim verifier for detecting duplicates
* when the verifier runs use the existing matching attributes finder code to find any duplciates
* create a record associating the new claim with these duplicates (`Duplicate belongs_to :source_claim, belongs_to :duplicate_claim`)
* replace the use of matching attribute finder in the app with looking at the new duplicates record for the given claim.
* this report can be simplified by simply joining claims to the new duplicate record.

Might need this query logic for the initial back fill on production but it can then be deleted.

What do you guys think?